frontend/pages/views/PhotographerListView.js

```diff
@@ -1,68 +1,97 @@
 import React from "react";
 import Footer from "../../components/Footer";
 import * as PropTypes from "prop-types";
-
+import {debounce} from "../../common";
 
 export class PhotographerListView extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            photographers: JSON.parse(this.props.photographers)
+            photographers: [],
         };
     }
 
     hrefFunc(number) {
         return `/photographer/${number}`;
-    };
+    }
 
     srcFunc(number) {
         return `${this.props.photoListDir}/${number}_photo.jpg`;
-    };
+    }
+
+    updatePhotographers(name) {
+        debounce(async () => {
+            const fetchedPhotographers = await this.fetchPhotographers(name);
+            this.setState({photographers: fetchedPhotographers});
+        }, 300)();
+    }
+
+    async fetchPhotographers(sq) {
+        try {
+            const res = await fetch(`/api/search_photographers?name=${sq}`);
+            return res.json();
+        } catch {
+            return [];
+        }
+    }
 
     getPhotoList() {
         const photoSize = [100, 100];
         return this.state.photographers.map((photog, k) => {
             return (
                 <li className="col-2 col-lg-2 one-photographer list-inline-item" key={k}>
                     <div className="child">
-                        <a key={k}
-                           href={this.hrefFunc(photog.number)}>
+                        <a key={k} href={this.hrefFunc(photog.number)}>
                             <img
                                 alt={photog.number}
                                 width={photoSize[0]}
-                                src={this.srcFunc(photog.number)}/>
+                                src={this.srcFunc(photog.number)}
+                            />
                         </a>
                         <p>{photog.name ? photog.name : "No Name"}</p>
                     </div>
                 </li>
             );
         });
+    }
 
+    componentDidMount() {
+        this.updatePhotographers("");
     }
 
     render() {
         return (
             <>
                 <div className="row">
-                    <p style={{
-                        fontSize: "30px",
-                        fontWeight: "600"
-                    }}>Photographers</p>
+                    <p
+                        style={{
+                            fontSize: "30px",
```

Comment:
Although this isn't the point of this PR, since you're working in this area -- this style info should get abstracted out into an SCSS file at some point. (doesn't have to happen in this PR, but before we finalize this module)